src/components/PollSection/Components/PollOptionList.tsx

```diff
@@ -0,0 +1,142 @@
+  FC,
+  memo,
+  useCallback,
+  useMemo,
+  useState,
+  useEffect,
+} from 'react';
+import { useStyles } from '../style';
+import { SvgXml } from 'react-native-svg';
+import { radioOff, radioOn } from '../../../svg/svg-xml-list';
+import PollBar from './PollBar';
+
+interface IPollOptionList {
+  options: Amity.PollAnswer[];
+  answerType: Amity.PollAnswerType;
+  isAlreadyVoted: boolean;
+  isPollClosed: boolean;
+  totalVote: number;
+  onSubmit: (selectedOption: string[]) => void;
+}
+
+const PollOptionList: FC<IPollOptionList> = ({
+  options: data = [],
+  answerType,
+  isAlreadyVoted,
+  isPollClosed,
+  totalVote,
+  onSubmit,
+}) => {
+  const styles = useStyles();
+  const [options, setOptions] = useState(data.slice(0, 2) || []);
+  const [selectedOption, setSelectedOption] = useState<string[]>([]);
+  const hiddenOptions = data?.length - [...options]?.length || 0;
+  const showedAllOptions = hiddenOptions === 0;
+  const isSingleOption = answerType === 'single';
+
+  useEffect(() => {
+    setOptions(data.slice(0, 2));
+  }, [data]);
+
+  const onPressMoreOption = useCallback(() => {
+    setOptions([...data]);
+  }, [data]);
+
+  const onPressOption = useCallback(
+    (optionId: string) => {
+      if (isSingleOption) return setSelectedOption([optionId]);
+      const currentSelectedOptions = [...selectedOption];
+      const index = currentSelectedOptions.findIndex((id) => id === optionId);
+      if (index !== -1) {
+        currentSelectedOptions.splice(index, 1);
+      } else {
+        currentSelectedOptions.push(optionId);
+      }
+      setSelectedOption([...currentSelectedOptions]);
+    },
+    [isSingleOption, selectedOption]
+  );
+
+  const renderSubmitButton = useMemo(() => {
+    if (!isPollClosed && !isAlreadyVoted) {
+      return (
+        <TouchableOpacity
+          onPress={() => onSubmit(selectedOption)}
+          style={styles.submitBtn}
+        >
+          <Text style={styles.submit}>Submit</Text>
+        </TouchableOpacity>
+      );
+    } else return null;
+  }, [isPollClosed, isAlreadyVoted, styles, onSubmit, selectedOption]);
+
+  const renderViewAllOptions = useMemo(() => {
+    const resultText =
+      isPollClosed || isAlreadyVoted
+        ? `View full results`
+        : `${hiddenOptions} more options`;
+    return (
+      <TouchableOpacity
+        onPress={onPressMoreOption}
+        style={styles.moreOptionsBtn}
+      >
+        <Text>{resultText}</Text>
+      </TouchableOpacity>
+    );
+  }, [
+    hiddenOptions,
+    isPollClosed,
+    isAlreadyVoted,
+    onPressMoreOption,
+    styles.moreOptionsBtn,
+  ]);
+
+  return (
+    <>
+      {options.map((option) => {
+        const selectedItem = selectedOption.includes(option.id);
+        const selectedIcon = selectedItem ? radioOn() : radioOff();
+        const selectedOptionContainerStyle: ViewStyle =
+          selectedItem && styles.selectedOptionContainer;
+        const length = (option.voteCount / totalVote) * 100;
```

Comment:
imo, it would be better to make sure that `length` always `number`

Reply:
Updated